lib/image.js

```diff
@@ -1,12 +1,71 @@
-"use strict";
-var Promise = require('bluebird');
+'use strict';
+var Cesium = require('cesium');
 var fs = require('fs-extra');
 var path = require('path');
+var PNG = require('pngjs').PNG;
+var Promise = require('bluebird');
 
 var fsReadFile = Promise.promisify(fs.readFile);
 
+var WebGLConstants = Cesium.WebGLConstants;
+
 module.exports = loadImage;
 
+/**
+ * Load an image file and get information about it.
+ *
+ * @param {String} imagePath Path to the image file.
+ * @returns {Promise} A promise resolving to the image information, or undefined if the file doesn't exist.
+ *
+ * @private
+ */
+function loadImage(imagePath) {
+    return fsReadFile(imagePath)
+        .then(function(data) {
+            var extension = path.extname(imagePath);
+            var uriType = getUriType(extension);
+            var uri = uriType + ';base64,' + data.toString('base64');
+
+            var info = {
+                transparent : false,
+                data : data,
+                uri : uri,
+                format : getFormat(3),
+                extension : extension
+            };
+
+            if (extension === '.png') {
+                // Color type is encoded in the 25th bit of the png
+                var colorType = data[25];
+                var channels = getChannels(colorType);
+                info.format = getFormat(channels);
+
+                if (channels === 4) {
+                    // Need to do a finer grained check over the pixels to see if the image is actually transparent
```

Comment:
As discussed offline, a lot of images may be 4-channel PNGs that don't have any transparent pixels. I wonder if it's okay to just make transparency a flag option, since a full pixel level check might be kind of slow.